ruma-identity-service-api/src/authentication/register/v2.rs

```diff
@@ -0,0 +1,40 @@
+//! [POST /_matrix/identity/v2/account/register](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-account-register)
+
+use std::time::Duration;
+
+use ruma_api::ruma_api;
+use ruma_identifiers::ServerNameBox;
+
+ruma_api! {
+
+    metadata: {
+        description: "Exchanges an OpenID token from the homeserver for an access token to access the identity server.",
+        method: POST,
+        name: "register_account",
+        path: "/_matrix/identity/v2/account/register",
+        authentication: None,
+        rate_limited: false,
+    }
+
+    request: {
+        /// An access token the consumer may use to verify the identity of the
+        /// person who generated the token. This is given to the federation API
+        /// GET /openid/userinfo to verify the user's identity.
+        pub access_token: String,
```

Comment:
We can use borrowing in the request here, right?

Reply:
You mean `pub access_token: &'a str`? Sure, if you say so! :laughing: 

I didn't track super well with those changes this summer. What are the criteria for using borrowed types? Maybe I'll add them to the contributing guide.

Reply:
Criteria: String stuff in `request`. Not possible in `response` yet. Some structs may also have to be converted, I'll try writing up more specific criteria soon.

Reply:
Okay, so it's still rather vague but:

* strings should be `&'a str`
* identifiers should be `&'a IdentifierType`
* `Vec`s should be `&'a [_]`. The inner type should *probably* not be touched (still a bit unclear)
* `BTreeMap`s and such should not be touched (probably)
* structs that are only ever used inside request blocks (no response blocks or events) should be lifetime-parameterized and apply the same rules to their fields; i.e. `my_field: MyStruct` => `my_field: MyStruct<'a>` (unless `MyStruct` has no string / ident fields)

Reply:
Adding to contributor's guide in #382.